test/src/crdt/crdt_datastore_test.cpp

```diff
@@ -97,10 +111,12 @@ namespace sgns::crdt
     EXPECT_OUTCOME_TRUE_1(transaction.Put(newKey1, buffer1));
     EXPECT_OUTCOME_TRUE_1(transaction.Put(newKey2, buffer2));
     EXPECT_OUTCOME_TRUE_1(transaction.Put(newKey3, buffer3));
-    EXPECT_OUTCOME_TRUE_1(transaction.Remove(newKey2));
+    // this won't work as part of the same atomic transaction, because the Remove looks for the existing key
+    // to create the delta, and since it's queued in the atomic transaction, it doesn't find key2
+    //EXPECT_OUTCOME_TRUE_1(transaction.Remove(newKey2));
     EXPECT_OUTCOME_TRUE_1(transaction.Commit());
     EXPECT_OUTCOME_EQ(crdtDatastore_->HasKey(newKey1), true);
-    EXPECT_OUTCOME_EQ(crdtDatastore_->HasKey(newKey2), false);
+    //EXPECT_OUTCOME_EQ(crdtDatastore_->HasKey(newKey2), false);
```

Comment:
I believe we should add a test for removal with atomic transaction, just to maintain test coverage.
```suggestion
    AtomicTransaction transactionRemoveKey2 = AtomicTransaction(crdtDatastore_);
    EXPECT_OUTCOME_TRUE_1(transactionRemoveKey2.Remove(newKey2));
    EXPECT_OUTCOME_TRUE_1(transactionRemoveKey2.Commit());
    EXPECT_OUTCOME_EQ(crdtDatastore_->HasKey(newKey1), true);
    EXPECT_OUTCOME_EQ(crdtDatastore_->HasKey(newKey2), false);
    EXPECT_OUTCOME_EQ(crdtDatastore_->HasKey(newKey3), true);
```